crates/rnote-compose/src/builders/penpathmodeledbuilder.rs

```diff
@@ -189,19 +186,17 @@ n_steps exceeds configured max outputs per call."
         }
         self.last_element_time = now;
 
-        let modeler_input = ModelerInput::new(
+        let modeler_input = ModelerInput {
             event_type,
-            (element.pos[0] as f32, element.pos[1] as f32),
-            now.duration_since(self.start_time).as_secs_f64(),
-            element.pressure as f32,
-            0.0,
-            0.0,
-        );
+            pos: (element.pos[0] as f32, element.pos[1] as f32),
+            time: now.duration_since(self.start_time).as_secs_f64(),
+            pressure: element.pressure as f32,
+        };
 
         match self.stroke_modeler.update(modeler_input) {
             Ok(results) => self.buffer.extend(results.into_iter().map(|r| {
-                let pos = r.pos();
-                let pressure = r.pressure();
+                let pos = r.pos;
+                let pressure = r.pressure;
                 Element::new(na::vector![pos.0 as f64, pos.1 as f64], pressure as f64)
             })),
             Err(e) => tracing::error!("Updating stroke modeler with element failed, Err: {e:?}"),
```

Comment:
I feel like changing the error types in the modeler could make this part completely redundant.
Just call the update on the model, match on the results and different errors messages (maybe an enum type for errors), this way we don't test twice that the events are not a duplicate, that the time delta is positive and that the number of steps is not too large.

Reply:
Yeah, that's what I meant with strongly typed errors in `ink-stroke-modeler`. But I also want `riir` to land soon, it can be a follow-up